v0/destinations/google_ads_enhanced_conversion/util.js

```diff
@@ -0,0 +1,148 @@
+const { get, set } = require("lodash");
+const sha256 = require("sha256");
+const { httpSend } = require("../../../adapters/network");
+const { isHttpStatusSuccess } = require("../../util/index");
+const { TRANSFORMER_METRIC } = require("../../util/constant");
+const ErrorBuilder = require("../../util/error");
+const {
+  DISABLE_DEST,
+  REFRESH_TOKEN
+} = require("../../../adapters/networkhandler/authConstants");
+const Cache = require("../../util/cache");
+
+const conversionActionIdCache = new Cache(24 * 60 * 60);
+
+const {
+  processAxiosResponse
+} = require("../../../adapters/utils/networkUtils");
+const { BASE_ENDPOINT } = require("./config");
+
+/**
+ *  This function is used for collecting the conversionActionId using the conversion name
+ * @param {*} method
+ * @param {*} headers
+ * @param {*} params
+ * @returns
+ */
+
+const getConversionActionId = async (method, headers, params) => {
+  const conversionActionIdKey = sha256(
+    params.event + params.customerId
+  ).toString();
+  return conversionActionIdCache.get(conversionActionIdKey, async () => {
+    const data = {
+      query: `SELECT conversion_action.id FROM conversion_action WHERE conversion_action.name = '${params.event}'`
+    };
+    const requestBody = {
+      url: `${BASE_ENDPOINT}/${params.customerId}/googleAds:searchStream`,
+      data,
+      headers,
+      method
+    };
+    const response = await httpSend(requestBody);
+    return response;
+  });
+};
+
+/**
+ * This function is responsible for collecting the conversionActionId
+ * and calling the enhanced conversion.
+ * data to customer list.
+ * @param {*} request
+ * @returns
+ */
+const ProxyRequest = async request => {
+  const { body, method, endpoint, params } = request;
+  const { headers } = request;
+
+  const conversionActionIdResponse = await getConversionActionId(
+    method,
+    headers,
+    params
+  );
+  if (
+    !conversionActionIdResponse.success &&
+    !isHttpStatusSuccess(conversionActionIdResponse.response.response.status)
+  ) {
+    // eslint-disable-next-line prefer-destructuring
+    set(
```

Comment:
Why not return a new response rather than mutating the existing one, it lowers risk

Reply:
Here, in enhanced conversion, google is returning the data as an array but for remarketing list, it was an object. So mutating it.

Reply:
Updated this.